artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -343,6 +353,10 @@ private void checkNumberOfPages() {
       if (!isBelowPageLimitBytes()) {
          this.pageFull = true;
          ActiveMQServerLogger.LOGGER.pageFullMaxBytes(storeName, numberOfPages, estimatedMaxPages, pageLimitBytes, pageSize);
+      } else {
+         if (cursorProvider != null) {
+            cursorProvider.checkClearPageLimit();
+         }
```

Comment:
It feels wrong for the wider method called checkNumberOfPages() to also be doing this, when its more aimed at the message count stuff. Especially as you already call this directly if the message limit changes.

EDIT: see suggestion from earlier comment.